File: 프로그래머스/1/42748. K번째수/K번째수.py
def solution(array, commands):
    result = []
    for i, j, k in commands:
        result.append(sorted(array[i-1:j])[k-1])
    return result

# 테스트 예제
array = [1, 5, 2, 6, 3, 7, 4]
commands = [[2, 5, 3], [4, 4, 1], [1, 7, 3]]
print(solution(array, commands))  # [5, 6, 3]

File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        int[] answer = new int[commands.length];
        
        int n = 0 ;
        for(int[] c : commands){
            int i = c[0];
            int j = c[1];
            int k = c[2];
            int[] li = Arrays.copyOfRange(array,i-1,j);
            Arrays.sort(li);
            answer[n] = li[k-1];
            n+=1 ;
        }
        
        return answer;
    }
}